api-server/src/server/boot/challenge.js

```diff
@@ -355,7 +355,18 @@ export function isValidChallengeCompletion(req, res, next) {
     log('challengeType', challengeType, isNumeric(challengeType));
     return res.status(403).json(isValidChallengeCompletionErrorMsg);
   }
-  if ('solution' in req.body && !isURL(solution)) {
+  // If `backEndProject`:
+  // - `solution` needs to exist, but does not have to be valid URL
+  // - `githubLink` needs to exist and be valid URL
+  if (challengeType === challengeTypes.backEndProject) {
+    if (!solution || !githubLink) {
+      log('isObjectId', id, ObjectID.isValid(id));
+      return res.status(403).json(isValidChallengeCompletionErrorMsg);
+    } else if (!isURL(githubLink)) {
+      log('isObjectId', id, ObjectID.isValid(id));
+      return res.status(403).json(isValidChallengeCompletionErrorMsg);
+    }
+  } else if ('solution' in req.body && !isURL(solution)) {
```

Comment:
The comment above this says, `` `solution` needs to exist, but does not have to be valid URL`` - this appears to check that it is valid. Just want to make sure that's intended. I think it's fine.

Reply:
> If `backEndProject`, `solution` needs to exist, but does not have to be valid. Hence:

```js
if (challengeType === challengeTypes.backEndProject)
```